prio/cli/commands.py

```diff
@@ -0,0 +1,318 @@
+import click
+import array
+import json
+import os
+from base64 import b64decode, b64encode
+from uuid import uuid4
+
+from .. import libprio
+from .options import (
+    data_config,
+    server_config,
+    input_1,
+    input_2,
+    output_1,
+    output_2,
+    public_key,
+)
+
+
+@click.command()
+def shared_seed():
+    """Generate a shared server secret in base64."""
+    seed = libprio.PrioPRGSeed_randomize()
+    click.echo(b64encode(seed))
+
+
+@click.command()
+def keygen():
+    """Generate a curve25519 key pair as json."""
+    private, public = libprio.Keypair_new()
+    private_hex = libprio.PrivateKey_export_hex(private).decode("utf-8")[:-1]
+    public_hex = libprio.PublicKey_export_hex(public).decode("utf-8")[:-1]
+    data = json.dumps({"private_key": private_hex, "public_key": public_hex})
+    click.echo(data)
+
+
+@click.command()
+@data_config
+@public_key
+@input_1
+@output_2
+def encode_shares(
+    batch_id,
+    n_data,
+    public_key_internal,
+    public_key_external,
+    input,
+    output_a,
+    output_b,
+):
+    public_key_internal = libprio.PublicKey_import_hex(public_key_internal)
+    public_key_external = libprio.PublicKey_import_hex(public_key_external)
+    config = libprio.PrioConfig_new(
+        n_data, public_key_internal, public_key_external, batch_id
+    )
+
+    with open(input, "r") as f:
+        data = map(json.loads, f.readlines())
+
+    name = os.path.basename(input)
+    path_a = os.path.join(output_a, name)
+    path_b = os.path.join(output_b, name)
+    with open(path_a, "w") as fp_a, open(path_b, "w") as fp_b:
+        for datum in data:
+            share_a, share_b = libprio.PrioClient_encode(config, bytes(datum))
+            uid = str(uuid4())
+            json.dump({"id": uid, "payload": b64encode(share_a).decode()}, fp_a)
+            fp_a.write("\n")
+            json.dump({"id": uid, "payload": b64encode(share_b).decode()}, fp_b)
+            fp_b.write("\n")
+
+
+@click.command()
+@data_config
+@server_config
+@public_key
+@input_1
+@output_1
+def verify1(
+    batch_id,
+    n_data,
+    server_id,
+    private_key,
+    shared_secret,
+    public_key_internal,
+    public_key_external,
+    input,
+    output,
+):
+    """Decode a batch of shares"""
+    click.echo("Running verify1")
+
+    private_key = libprio.PrivateKey_import_hex(private_key, public_key_internal)
+    public_key_internal = libprio.PublicKey_import_hex(public_key_internal)
+    public_key_external = libprio.PublicKey_import_hex(public_key_external)
+    server_id = libprio.PRIO_SERVER_A if server_id == "A" else libprio.PRIO_SERVER_B
+    shared_secret = b64decode(shared_secret)
```

Comment:
It seems like this block tends to be duplicated across many of the commands. Could you write some small helper functions to do this work, that could be shared? Also probably better not to modify input variables, but rather to transform them.

Could you do something like this:

```py
def import_keys(private_key_hex, public_key_internal_hex, public_key_external_hex):
    return (
        libprio.PrivateKey_import_hex(private_key_hex, public_key_internal_hex),
        libprio.PublicKey_import_hex(public_key_internal_hex),
        libprio.PublicKey_import_hex(public_key_external_hex)
    )
...
(private_key, public_key_internal, public_key_external) = import_keys(private_key_hex, public_key_internal_hex, public_key_external_hex)
```
```

Reply:
Done, I've avoided modifying input variables and did some refactoring.